src/rules/sat_3sat.jl

```diff
@@ -0,0 +1,136 @@
+"""
+The reduction result of a general SAT problem to a 3-SAT problem.
+
+### Fields
+- `sat_source::Satisfiability{GT, T}`: the source general SAT problem.
+"""
+struct ReductionSATTo3SAT{T}
+    sat_source::Satisfiability{T}
+    sat_target::Satisfiability{T}
+    new_var_map::Dict{Symbol, Symbol}
+    inverse_new_var_map::Dict{Symbol, Symbol}
+end
+target_problem(res::ReductionSATTo3SAT) = res.sat_target
+
+function reduceto(::Type{<:Satisfiability}, sat_source::Satisfiability)
```

Comment:
Maybe you want to reduce the problem to KSatisfiability problem?

Reply:
Resolved: Separate the reduction into SAT=>3-SAT and k-SAT=>SAT